pkg/aspect/attrgenmgr.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2017 the Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aspect
+
+import (
+	"github.com/golang/glog"
+	rpc "github.com/googleapis/googleapis/google/rpc"
+
+	"istio.io/mixer/pkg/adapter"
+	apb "istio.io/mixer/pkg/aspect/config"
+	"istio.io/mixer/pkg/attribute"
+	"istio.io/mixer/pkg/config"
+	"istio.io/mixer/pkg/config/descriptor"
+	cpb "istio.io/mixer/pkg/config/proto"
+	"istio.io/mixer/pkg/expr"
+	"istio.io/mixer/pkg/status"
+)
+
+type (
+	attrGenMgr  struct{}
+	attrGenExec struct {
+		aspect adapter.AttributesGenerator
+		params *apb.AttributeGeneratorsParams
+	}
+)
+
+func newAttrGenMgr() PreprocessManager {
+	return &attrGenMgr{}
+}
+
+func (attrGenMgr) Kind() config.Kind {
+	return config.AttributesKind
+}
+
+func (attrGenMgr) DefaultConfig() (c config.AspectParams) {
+	// NOTE: The default config leads to the generation of no new attributes.
+	return &apb.AttributeGeneratorsParams{}
+}
+
+func (attrGenMgr) ValidateConfig(c config.AspectParams, v expr.Validator, df descriptor.Finder) (cerrs *adapter.ConfigErrors) {
```

Comment:
Basically what we discussed out of band of the github review: a `Manifest` of some type that lists, with `AttributeDescriptor`s, the attributes that this adapter produces. I'm fine keeping `map<string, string>` for now since we'll change it later.